source/extensions/filters/http/grpc_json_transcoder/json_transcoder_filter.cc

```diff
@@ -757,6 +762,25 @@ bool JsonTranscoderFilter::checkIfTranscoderFailed(const std::string& details) {
   return false;
 }
 
+bool JsonTranscoderFilter::checkAndRejectIfResponseTranscoderFailed() {
+  const auto& response_status = transcoder_->ResponseStatus();
+  if (!response_status.ok()) {
+    ENVOY_LOG(debug, "Transcoding response error {}", response_status.ToString());
+    error_ = true;
+    encoder_callbacks_->sendLocalReply(
+        Http::Code::InternalServerError,
```

Comment:
502 Bad Gateway is more appropriate for this case?

>    The 502 (Bad Gateway) status code indicates that the server, while
   acting as a gateway or proxy, received an invalid response from an
   inbound server it accessed while attempting to fulfill the request.